lib/sax.js

```diff
@@ -962,6 +980,42 @@
     return result
   }
 
+  var parsers = {}
```

Comment:
This creates a mapping from each `state` to the correct function. Because of the number of branches in the original `switch` statement, this approach leads to performance gains due to significantly fewer compares/jumps.